pandas/core/groupby/generic.py

```diff
@@ -1362,13 +1358,23 @@ def _transform_general(
     @Appender(_transform_template)
     def transform(self, func, *args, engine=None, engine_kwargs=None, **kwargs):
 
+        if maybe_use_numba(engine):
+            if not callable(func):
```

Comment:
*maybe* this check should actually be in _transform_with_numba to keep DRY (you have it above as well)